libcontainer/intelrdt/intelrdt.go

```diff
@@ -585,93 +613,117 @@ func (m *IntelRdtManager) Destroy() error {
+				return nil, nil
 			}
-		}
 
-		// The L3 cache schema in 'container_id' group
-		for _, schema := range schemaStrings {
-			if strings.Contains(schema, "L3") {
-				stats.L3CacheSchema = strings.TrimSpace(schema)
+			rootPath, err := getIntelRdtRoot()
+			if err != nil {
+				return nil, err
 			}
-		}
-	}
+			// The read-only L3 cache and memory bandwidth schemata in root
+			tmpRootStrings, err := getIntelRdtParamString(rootPath, "schemata")
+			if err != nil {
+				return nil, err
+			}
+			schemaRootStrings := strings.Split(tmpRootStrings, "\n")
 
-	if IsMbaEnabled() {
-		// The read-only memory bandwidth information
-		memBwInfo, err := getMemBwInfo()
-		if err != nil {
-			return nil, err
-		}
-		stats.MemBwInfo = memBwInfo
+			// The L3 cache and memory bandwidth schemata in 'container_id' group
 
-		// The read-only memory bandwidth information
-		for _, schemaRoot := range schemaRootStrings {
-			if strings.Contains(schemaRoot, "MB") {
-				stats.MemBwSchemaRoot = strings.TrimSpace(schemaRoot)
+			tmpStrings, err := getIntelRdtParamString(containerPath, "schemata")
+			if err != nil {
+				return nil, err
 			}
-		}
+			schemaStrings := strings.Split(tmpStrings, "\n")
 
-		// The memory bandwidth schema in 'container_id' group
-		for _, schema := range schemaStrings {
-			if strings.Contains(schema, "MB") {
-				stats.MemBwSchema = strings.TrimSpace(schema)
+			if IsCatEnabled() {
+				// The read-only L3 cache information
+				l3CacheInfo, err := getL3CacheInfo()
+				if err != nil {
+					return nil, err
+				}
+				stats.L3CacheInfo = l3CacheInfo
+
+				// The read-only L3 cache schema in root
+				for _, schemaRoot := range schemaRootStrings {
+					if strings.Contains(schemaRoot, "L3") {
+						stats.L3CacheSchemaRoot = strings.TrimSpace(schemaRoot)
+					}
+				}
+
+				// The L3 cache schema in 'container_id' group
+				for _, schema := range schemaStrings {
+					if strings.Contains(schema, "L3") {
+						stats.L3CacheSchema = strings.TrimSpace(schema)
+					}
+				}
+			}
+
+			if IsMbaEnabled() {
+				// The read-only memory bandwidth information
+				memBwInfo, err := getMemBwInfo()
+				if err != nil {
+					return nil, err
+				}
+				stats.MemBwInfo = memBwInfo
+
+				// The read-only memory bandwidth information
+				for _, schemaRoot := range schemaRootStrings {
+					if strings.Contains(schemaRoot, "MB") {
+						stats.MemBwSchemaRoot = strings.TrimSpace(schemaRoot)
+					}
+				}
+
+				// The memory bandwidth schema in 'container_id' group
+				for _, schema := range schemaStrings {
+					if strings.Contains(schema, "MB") {
+						stats.MemBwSchema = strings.TrimSpace(schema)
+					}
+				}
 			}
 		}
-	}
 
-	err = getMonitoringStats(containerPath, stats)
-	if err != nil {
-		return nil, err
+		return stats, nil
+
+	case MON:
+		path := m.GetPath()
+
+		err := getMonitoringStats(path, stats)
+		if err != nil {
+			return nil, err
+		}
+		return stats, nil
 	}
 
-	return stats, nil
+	return nil, nil
```

Comment:
It looks strange but I see that `nil, nil` is returned when type was different than CTRL_MON and different than MON so maybe it will be useful to add log with proper information (maybe info level log).

Reply:
It will be easier to return the error.

Reply:
I also provided a unit test for that.